pkg/ipam/types.go

```diff
@@ -114,3 +115,17 @@ func (ipam *IPAM) DebugStatus() string {
 	ipam.allocatorMutex.RUnlock()
 	return str
 }
+
+// GetVpcCIDRs returns all the CIDRs associated with the VPC this node belongs to.
+// This works only cloud provider IPAM modes and returns nil for other modes.
+func (ipam *IPAM) GetVpcCIDRs() (vpcCIDRs []*cidr.CIDR) {
+	primary, secondary := deriveVpcCIDRs(sharedNodeStore.ownNode)
```

Comment:
I think this needs some locking before it can access the node

Reply:
Yes, missed that. Added an `RLock` on  `sharedNodeStore `